libs/component-library/src/lib/radio-button/RadioButton.tsx

```diff
@@ -0,0 +1,228 @@
+import { createBox, useTheme } from '@shopify/restyle';
+import * as React from 'react';
+import {
+  Animated,
+  View,
+  StyleSheet,
+  TouchableOpacity,
+  TouchableOpacityProps,
+} from 'react-native';
+import { FxTheme } from '../theme/theme';
+
+import type { $RemoveChildren } from './../types';
+
+import { useRadioButtonContext } from './RadioButtonGroup';
+import { handlePress, isChecked } from './utils';
+
+const TouchableBox = createBox<FxTheme, TouchableOpacityProps>(
+  TouchableOpacity
+);
+
+type Props = $RemoveChildren<typeof TouchableBox> & {
+  /**
+   * Value of the radio button
+   */
+  value: string;
+  /**
+   * Status of radio button.
+   */
+  status?: 'checked' | 'unchecked';
+  /**
+   * Whether radio is disabled.
+   */
+  disabled?: boolean;
+  /**
+   * Function to execute on press.
+   */
+  onPress?: () => void;
```

Comment:
is `onPress and `disabled` needed here?, Don't these already come in from the `TouchableOpacityProps`?

Reply:
So the `onPress` here doesn't actually receive the `GestureResponderEvent` but is rather executed in the internal `onPress`. I will update the comment to say in `onPress`

Reply:
For the `disabled` I wanted to update the jsdoc comments to say radio